src/v2/Components/__stories__/Molecules.story.tsx

```diff
@@ -0,0 +1,200 @@
+import {
+  Box,
+  Text,
+  GridColumns,
+  Column,
+  Button,
+  Spacer,
+  ProgressDots,
+} from "@artsy/palette"
+import { themeGet } from "@styled-system/theme-get"
+import React from "react"
+import styled from "styled-components"
+import { FullBleed } from "../FullBleed"
+
+export default {
+  title: "Molecules/Kitchen Sink",
+  parameters: {
+    layout: "fullscreen",
+  },
+}
+
+const AppContainer = styled(Box)`
+  max-width: ${themeGet("breakpoints.lg")};
+`
+
+AppContainer.defaultProps = {
+  mx: "auto",
+  px: [2, 4],
+}
+
+const Page: React.FC = ({ children }) => {
+  return <AppContainer>{children}</AppContainer>
+}
+
+const MetadataExample = () => {
+  return (
+    <>
+      {/* TODO: Mixin a textTransform prop to text */}
+      <Text variant="xs" style={{ textTransform: "uppercase" }} mb={1}>
+        Category Title
+      </Text>
+
+      <Text variant="xl">Section Title or Headline</Text>
+
+      <Text variant="xl" color="black60" mb={1}>
+        Secondary Text
+      </Text>
+
+      <Text variant="lg" mb={1}>
+        Sub Headline
+      </Text>
+
+      <Text variant="sm" mb={4}>
+        Knausgaard vexillologist shabby chic, bitters vape vaporware live-edge
+        mixtape ennui raclette
+      </Text>
+
+      {/* TODO: Support n-column grids with a context provider as an alternative to sub-grid support? */}
```

Comment:
![](https://static.damonzucconi.com/_capture/wwyDJOSM.png)

[Sub-grid won't be ready anytime soon.](https://caniuse.com/css-subgrid)

What I'm thinking is: I can add support for n-column layouts in the `GridColumns` component and then every instance of `Column` can also be a context provider which sets the column count for any sub-grid. So something like

```tsx
<Grid> // default 12-column
  ...
  <Column span={6}>
    ...
    <SubGrid> // equivalent to <Grid columns={6}> - pulls in column count from outer column context
      <Column span={4}>
        <Button width="100%>...</Button>
      </Column>
    </SubGrid>
  </Column>
<Grid>

Reply:
Though we'd also want to enforce that any sub-grid is divisible by the outermost parent grid 😬

Reply:
It comes down to how people will use this in practice. One dev might do 
```
<Grid>
  <Col ...>
     <Grid>
        <Col ...>
```
And another dev might just do 
```
<Box width="50%">
  <Button>
</Box>
```

It might be worth putting in the time to engineer a solution tho if you can imagine folks will use this approach over some of the more literal ones.

Reply:
Oh sure; but in this instance for example: the button was roughly equivalent to 60% of the container. These designs are pretty specific about buttons occupying column widths so it'd be nice to actually be able to do it for real (for instance span={6} isn't exactly 50% because of the gutter spacing)

Reply:
Ahh, yeah, this is generally hard to do. An idealistic part of me likes the idea of completely flattening layouts to a singular component that describes areas and their dimensions. Components that are placed into those areas wouldn't know nor care about their layout. 

https://github.com/kettanaito/atomic-layout is a library that implements kind of what I'm describing. You end up having to think about the layout more at the page level though so I'm not sure if that's better or worse.